src/shapes/Text/Text.ts

```diff
@@ -1690,11 +1703,11 @@ export class Text<
           this.setPathInfo();
         }
         needsDims =
-          needsDims || this._dimensionAffectingProps.indexOf(_key) !== -1;
+          needsDims || this.constructor.textLayoutProperties.includes(_key);
         isAddingPath = isAddingPath || _key === 'path';
       }
     } else {
-      needsDims = this._dimensionAffectingProps.indexOf(key) !== -1;
+      needsDims = this.constructor.textLayoutProperties.includes(key);
```

Comment:
This thing here remind me of te issue we have with side effects.
sometimes those fit in _set, but if those are slightly expensive we want to run them only once for set.
But this bring us to handle in set both single and multi key use case.
@ShaMan123  should check your side effects pr solves this issue with 10 lines of code or find a valid way to remove those with a little footprint.

Reply:
don't think it is 10 lines but it solves this bad design